src/@utils/order.ts

```diff
@@ -20,6 +21,27 @@ import {
 } from '../../app.config'
 import { toast } from 'react-toastify'
 
+async function initializeProvider(
+  asset: AssetExtended,
+  accountId: string,
+  providerFees?: ProviderFees
+): Promise<ProviderInitialize> {
+  if (!providerFees) {
```

Comment:
super small thing, but we mostly utilize return early pattern to prevent excessive nesting. So in this case just turn around logic to remove the first level nesting by doing `if (providerFees) return`.

For the rest, just noting here as I'm not sure we want to touch that mega `@utils/order` further, as changes are hard to test.

Another problem, but I get why it's here as it was just hacked into original code: accepting a parameter which then is only used to check if method should run seems unnecessary. Cause why is the method even called in the first place when existence of `providerFees` is known before calling it? Better pattern here would be to only call this method when necessary, which is only when `providerFees` are not present.

And finally, this method should always return something as it's kinda only luck leading to this not breaking as we have `providerFees || initializeData.providerFee` further down. Furthermore, if no `providerFees` are passed AND `initializeProvider()` returns empty `initializeData`, the app breaks. Solution could be optional chaining with `initializeData?.providerFee` and make `initializeProvider()` always return something